src/pages/Calculator.tsx

```diff
@@ -272,124 +260,151 @@ export const Calculator = (): React.ReactElement => {
-                    <BaseTransmissionRateMessage
-                      interaction={calculatorData.interaction}
-                      messageKey="calculator.one_time_interaction_risk_message"
-                    />
-                  ) : null}
-                  {calculatorData.interaction === 'repeated' ? (
-                    <BaseTransmissionRateMessage
-                      interaction={calculatorData.interaction}
-                      messageKey="calculator.repeated_interaction_risk_message"
-                    />
-                  ) : null}
-                  {calculatorData.interaction === 'partner' ? (
-                    <BaseTransmissionRateMessage
-                      interaction={calculatorData.interaction}
-                      messageKey="calculator.partner_interaction_risk_message"
-                    />
-                  ) : null}
-                  {calculatorData.interaction !== undefined &&
-                  calculatorData.interaction !== '' ? (
-                    <p>
-                      <Trans i18nKey="calculator.whats_next_interaction_risk_message">
-                        Lorem ipsum dolor <strong>sit amet</strong>
-                        (backed by{' '}
-                        <Link to="/paper/14-research-sources" target="_blank">
-                          research
-                        </Link>
-                        !)
-                      </Trans>
-                    </p>
-                  ) : null}
-                </div>
-                <Row>
-                  <Col xs="12" id="person-risk" className="calculator-params">
-                    <PersonRiskControls
-                      data={calculatorData}
-                      setter={setCalculatorData}
-                      repeatedEvent={repeatedEvent}
-                    />
-                  </Col>
-                  <Col xs="12" id="modifiers" className="calculator-params">
-                    <ActivityRiskControls
-                      data={calculatorData}
-                      setter={setCalculatorData}
-                      repeatedEvent={repeatedEvent}
-                    />
-                  </Col>
-                  <Col xs="12" id="vaccines" className="calculator-params">
-                    <VaccineSelector
-                      id="yourVaccine"
-                      header={t('calculator.precautions.your_vaccine_header')}
-                      label={t('calculator.precautions.your_vaccine_label')}
-                      dosesLabel={t(
-                        'calculator.precautions.your_vaccine_doses_label',
-                      )}
-                      variant="outline-cyan"
-                      current={{
-                        vaccineDoses: calculatorData.yourVaccineDoses,
-                        vaccineType: calculatorData.yourVaccineType,
+                      setSavedData={(newScenario: string): void => {
+                        let foundData:
```

Comment:
Why is this type so complicated? prepopulated is [`key: string]: PartialData`, so shouldn't foundData be of type `PartialData?`

Reply:
I copied it over from the existing `foundData`. `null` still needs to be there in case of invalid scenario names like `''` but yeah, it looks like it can't actually be a `calculatorData`.